src/OrbitGl/TrackManager.cpp

```diff
@@ -198,7 +198,6 @@ void TrackManager::UpdateVisibleTrackList() {
   // This function assumes we asked before for a update for the visible track list and that tracks
   // are already sorted (in sorted_tracks_).
```

Comment:
Isn't the second part of this comment outdated now?

Reply:
Not really. 

We are still assuming that sorted_tracks_ is sorted.
But that complete sorting could be invalidated because a new Track appear (and was inserted in all_tracks_).

Reply:
If I see this correctly, it can happen that `sorting_invalidated_ == true` if a new track was added async, and this will mean that there will be another round of sorting and updating visible tracks in the next frame, correct? 

I think we should either remove the comment or add `if (sorting_invalidated_) return;` at the start of this method (I'd prefer the latter). Maybe extend the comment to point out this will be handled in the next frame then. In the current way, the comment doesn't really help - so we assume the track list is sorted, maybe it's not though, but whatever, we don't care? :-)

Reply:
For the first option, I could agree on rephrase the comment and moved to definition of the function. 
```
// This function will filter tracks that are already sorted in sorted_tracks_.
```

The second option I don't like it in the current state of the code. If we return that, we should also mutex this function to assure that the order doesn't get invalidate during the filtering. And why we would want to do that?
This will be refactored in the future. Tracks won't be inserted from another thread when adding a new timer.

Reply:
Ok, sounds good to me as well.